### PR TITLE
Adding a user-agent header

### PR DESCRIPTION
**Pull request by masseyke**
Adding a simple user agent that contains just es-hadoop version information and hadoop or spark version information. For example:
```
elasticsearch-hadoop/8.5.0-SNAPSHOT spark/1.6.2
```

### CODE REVIEW

File: spark/core/src/main/scala/org/elasticsearch/spark/cfg/SparkSettings.java [hunk 1]
```diff
@@ -51,6 +53,13 @@ public SparkSettings(SparkConf cfg) {
         String appId = cfg.get("spark.app.id", "");
         String opaqueId = String.format(Locale.ROOT, "[spark] [%s] [%s] [%s]", user, appName, appId);
         this.setOpaqueId(opaqueId);
+        this.setUserAgent(String.format(
+                        Locale.ROOT,
+                        "elasticsearch-hadoop/%s spark/%s",
+                        Version.versionNumber(),
+                        SparkContext.getOrCreate(cfg).version()
```
Comment (jbaiera) [inline spark/core/src/main/scala/org/elasticsearch/spark/cfg/SparkSettings.java:60]: Is there a lighter way to get this version or are we sure that the SparkContext will already be present when getting it? Would creating the context here and not closing it out be problematic?

Comment (masseyke) [reply spark/core/src/main/scala/org/elasticsearch/spark/cfg/SparkSettings.java:60]: Yeah it does. I haven't figured out how to access that from Java yet though (it's not as straightforward as I'd have thought). I'm currently changing it to use a helper scala class to expose it to java.

**Review by jbaiera — APPROVED** [review]
LGTM, thanks for adding this

**Comment by masseyke** [comment]
This is part of #1992

**Pull request merged**